src/Microsoft.Kiota.Cli.Commons/Http/Headers/BaseHeadersStore.cs

```diff
@@ -87,14 +87,13 @@ public virtual bool AddHeaders(ICollection<string>? headers)
     {
         foreach (var headerLine in headers)
         {
-            var split = headerLine.Split('=',

```

Comment:
You could simply use the overload of the Split function to have a ceiling on the splits to achive the same. 
https://learn.microsoft.com/en-us/dotnet/api/system.string.split?view=net-6.0#system-string-split(system-char-system-int32-system-stringsplitoptions)

```cs
            var split = headerLine.Split('=',2,
    StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
```

Reply:
I prefer this version since the `Split` method allocates 2 strings and an array. In this situation, there's no array allocation. What do you think?

Reply:
I'm fine either way, but if we start arguing about memory complexity, then we should assign idx +1 to a variable instead of doing it twice.

Reply:
And we should pass a StringComparison.OrdinalIgnoreCase to the IndexOf method

Reply:
What's the reason behind adding StringComparison.OrdinalIgnoreCase for an = sign? Why do you think it's needed?

Reply:
I've added `StringComparison.Ordinal`. The idx + 1 calculation is usually optimized out by optimizing compilers. I'll need to look at the IL of C# to confirm. It usually is 1 cycle for both of them.